packages/babel-plugin-proposal-optional-chaining/src/index.js

```diff
@@ -113,7 +120,36 @@ export default declare((api, options) => {
               );
             }
           }
-
+          let replacement = replacementPath.node;
+          // Ensure (a?.b)() has proper `this`
+          if (
+            t.isMemberExpression(replacement) &&
+            (replacement.extra?.parenthesized ||
+              // if replacementPath.parentPath does not equal parentPath,
+              // it must be unwrapped from parenthesized expression.
+              replacementPath.parentPath !== parentPath) &&
+            parentPath.isCallExpression()
+          ) {
+            // `(a?.b)()` to `(a == null ? undefined : a.b.bind(a))()`
+            const { object } = replacement;
+            let baseRef;
+            if (!loose) {
+              // memoize the context object in non-loose mode
+              // `(a?.b.c)()` to `(a == null ? undefined : (_a$b = a.b).c.bind(_a$b))()`
```

Comment:
Do we need to check if there's a call expression in the path? `(o?.Foo().m)();` will double call, I think.